### PR TITLE
 Script to mass-update bot cache in all issues 

### PR DESCRIPTION
**Pull request by iarspider**
@smuzaffar for a single-use script, I see no reason to merge it. I will create and run Jenkins job after you review this.

### CODE REVIEW

**Comment by smuzaffar** [comment]
@iarspider , can you please automatically process all 
- cms organization from https://github.com/cms-sw/cms-bot/blob/master/gh-teams.py#L17  . Process all repos in these orgs except for `cms-sw` process only https://github.com/cms-sw/cms-bot/blob/master/cms_static.py#L15 repos?

File: mass-update-cache.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            cmsbuild_user = repo_config.CMSBUILD_USER
+            set_gh_user(cmsbuild_user)
+
+            if not getattr(repo_config, "RUN_DEFAULT_CMS_BOT", True):
+                sys.exit(0)
+            repo = gh.get_repo(repo_name)
+
+            cnt = 0
+
+            for issue in repo.get_issues():
```
Comment (smuzaffar) [inline mass-update-cache.py:62]: why process all issues? I would suggest to only process PRsand  sort the PRs for `updated` (https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests)

File: mass-update-cache.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                    )
+                    if re.match(ISSUE_SEEN_MSG, first_line):
+                        bot_cache = load_bot_cache_local(comment_msg)
+                        print("    Read bot cache from already seen comment:", comment)
+                        break
+                    elif re.match(CMSBOT_TECHNICAL_MSG, first_line):
+                        bot_cache = load_bot_cache_local(comment_msg)
+                        print("    Read bot cache from technical comment:", comment)
+                        break
+                if bot_cache and ("commits" not in bot_cache):
```
Comment (smuzaffar) [inline mass-update-cache.py:82]: I guess we would like to stop if there is no bot cache ... right? We do not want to process over 40K PRs , we only want to process those PRs where we have old bot cache

File: mass-update-cache.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                            "python3",
+                            "process-pull-request.py",
+                            "--force",
+                            "--dry-run" if args.dryrun else "",
+                            "--repository",
+                            args.repository,
+                            file=f,
+                        )
+
+        api_rate_limits(gh)
```
Comment (smuzaffar) [inline mass-update-cache.py:98]: move it under the for loop i.e. call it for every PR you process.

**Comment by smuzaffar** [comment]
@iarspider , looks good. I would suggest to add `--repo org/repo` option so that if passed then it only process that one repository. Then I would suggest to run it for `cms-data/L1Trigger-L1TCalorimeter` and see how many PR it has process and for whcih it suggested bot-cache update?

**Comment by iarspider** [comment]
The script can't find any PR to update bot cache in (among all repositories mentioned in [this comment](https://github.com/cms-sw/cms-bot/pull/2165#issuecomment-1930092787)), but I don't see a flaw in the logic... Double-checking now.

**Comment by iarspider** [comment]
In total, there are 6 PRs (all in cms-data) that need cache update.

**Comment by smuzaffar** [comment]
can you please report those 6 PRs here?

**Comment by iarspider** [comment]
cms-data/RecoEgamma-ElectronIdentification#28
cms-data/GeneratorInterface-EvtGenInterface#24
cms-data/Configuration-Generator#15
cms-data/PhysicsTools-NanoAOD#15
cms-data/RecoTracker-MkFit#13
cms-data/SimCalorimetry-EcalEBTrigPrimProducers#1

**Comment by smuzaffar** [comment]
cms-data PRs list looks good. what about cms-sw/cmsdist ?

**Comment by iarspider** [comment]
I have found a flaw in the script. Here is a revised list of all PRs that needs to be updated:

cms-sw/cmsdist#8982
cms-sw/cmsdist#8973
cms-sw/cms-bot#2163
cms-sw/cms-bot#2162
cms-sw/cms-bot#2160
cms-sw/cms-bot#2159
cms-sw/cms-bot#2156
cms-sw/root#198
cms-sw/root#196
cms-sw/root#197
cms-sw/root#192
cms-sw/root#195
cms-sw/root#194
cms-sw/root#193
cms-sw/root#191
cms-sw/root#190
cms-sw/cmssw-config#102
cms-sw/cmssw-config#101
cms-sw/cmssw-config#100
cms-sw/cmssw-config#98
cms-sw/cmssw-config#99
cms-sw/pkgtools#204
cms-sw/pkgtools#203
cms-sw/pkgtools#201
cms-sw/pkgtools#202
cms-sw/SCRAM#62
cms-sw/SCRAM#61
cms-sw/SCRAM#63
cms-sw/cmssw-osenv#13
cms-sw/cmssw-osenv#12
cms-sw/cms-git-tools#126
cms-sw/cms-git-tools#125
cms-sw/cms-common#10
cms-data/RecoEgamma-ElectronIdentification#28
cms-data/L1Trigger-L1TCalorimeter#26
cms-data/L1Trigger-L1TCalorimeter#25
cms-data/GeneratorInterface-EvtGenInterface#24
cms-data/RecoMET-METPUSubtraction#7
cms-data/Configuration-Generator#15
cms-data/RecoBTag-Combined#55
cms-data/RecoBTag-Combined#54
cms-data/RecoMuon-MuonIdentification#10
cms-data/PhysicsTools-NanoAOD#15
cms-data/RecoTauTag-TrainingFiles#13
cms-data/HeterogeneousCore-SonicTriton#4
cms-data/HeterogeneousCore-SonicTriton#3
cms-data/RecoTracker-MkFit#13
cms-data/RecoEcal-EgammaClusterProducers#5
cms-data/RecoEgamma-EgammaPhotonProducers#5
cms-data/SimCalorimetry-EcalEBTrigPrimProducers#1
cms-data/DataFormats-L1Scouting#2
cms-data/DataFormats-L1Scouting#1
cms-data/DataFormats-L1ScoutingRawData#2
cms-data/DataFormats-L1ScoutingRawData#1
cms-data/L1Trigger-L1CaloTrigger#1
cms-data/L1Trigger-L1TMuonEndCapPhase2#1
cms-externals/eigen-git-mirror#8

**Pull request merged**